sdcm/sct_config.py

```
@@ -305,6 +305,28 @@ def dict_or_str(value: dict | str | None) -> dict | None:
 
+def dict_or_str_or_int(value: dict | str | int | None) -> dict | int | None:
+    # bool is a subclass of int, so guard against it being silently accepted as 0/1.
+    if value is None or (isinstance(value, (dict, int)) and not isinstance(value, bool)):
+        return value
+    if isinstance(value, str):
+        try:
+            result = ast.literal_eval(value)  # "8" -> 8 ; "{'read': 8}" -> dict
+            if isinstance(result, (dict, int)) and not isinstance(result, bool):
+                return result
+        except Exception:  # noqa: BLE001
+            pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)